airflow/providers/google/cloud/hooks/datafusion.py

```diff
@@ -484,24 +516,31 @@ def start_pipeline(
         )
         runtime_args = runtime_args or {}
         # Unfortunately making the start call to CDAP does not return a run_id to poll for state.
-        # So we are adding a faux job id.
-        job_id = str(uuid.uuid4())
-        runtime_args[job_id_key] = job_id
+        # So we are adding a faux job id. Context link to relevant CDAP bug:
+        # https://issues.cask.co/browse/CDAP-7641
+        faux_job_id = str(uuid.uuid4())
+        runtime_args[job_id_key] = faux_job_id
 
         response = self._cdap_request(url=url, method="POST", body=runtime_args)
         if response.status != 200:
             raise AirflowException(
                 f"Starting a pipeline failed with code {response.status}"
             )
 
+        pipeline_id = self._get_pipeline_run_id(
+            pipeline_name=pipeline_name,
+            faux_pipeline_id=faux_job_id,
+            namespace=namespace,
+            instance_url=instance_url,
+        )
         self.wait_for_pipeline_state(
             success_states=[PipelineStates.RUNNING, PipelineStates.COMPLETED],
```

Comment:
nit we already have these defined in module a constant
```suggestion
            success_states=SUCCESS_STATES,
```